dipy/workflows/reconst.py

```diff
@@ -21,6 +21,165 @@
+        out_rtop : string, optional
+            Name of the rtop to be saved
+        out_lapnorm : string, optional
+            Name of the norm of laplacian signal to be saved
+        out_msd : string, optional
+            Name of the msd to be saved
+        out_qiv : string, optional
+            Name of the qiv to be saved
+        out_rtap : string, optional
+            Name of the rtap to be saved
+        out_rtpp : string, optional
+            Name of the rtpp to be saved
+        out_ng : string, optional
+            Name of the Non-Gaussianity to be saved
+        out_perng :  string, optional
+            Name of the Non-Gaussianity perpendicular to be saved
+        out_parng : string, optional
+            Name of the Non-Gaussianity parallel to be saved
+        laplacian_weighting :
+            Weighting value used in fitting the MAPMRI model in the laplacian
+            and both model types. (default: 0.05)
+        """
+        io_it = self.get_io_iterator()
+        for dwi, bval, bvec, out_rtop, out_lapnorm, \
+            out_msd, out_qiv, out_rtap, out_rtpp, out_ng,\
+            out_perng, out_parng in io_it:
+
+            logging.info('Computing MAPMRI metrics for {0}'.format(dwi))
+            img = nib.load(dwi)
+            data = img.get_data()
+            affine = img.affine
+            bvals, bvecs = read_bvals_bvecs(bval, bvec)
+
+            gtab = gradient_table(bvals=bvals, bvecs=bvecs,
+                                  small_delta=small_delta,
+                                  big_delta=big_delta, b0_threshold=b0_threshold)
+
+            if not save_metrics:
+                save_metrics = ['rtop', 'laplacian_signal', 'msd', 'qiv', 'rtap', 'rtpp',
+                                'ng', 'perng', 'parng']
+
+            radial_order = 6
+
+            if laplacian and positivity:
+                map_model_aniso = mapmri.MapmriModel(gtab, radial_order=radial_order,
+                                                     laplacian_regularization=True,
+                                                     laplacian_weighting=laplacian_weighting,
+                                                     positivity_constraint=True)
+                mapfit_aniso = map_model_aniso.fit(data)
+
+            elif positivity:
+
+                map_model_aniso = mapmri.MapmriModel(gtab, radial_order=radial_order,
+                                                     laplacian_regularization=False,
+                                                     positivity_constraint=True)
+                mapfit_aniso = map_model_aniso.fit(data)
+
+            elif laplacian:
+                map_model_aniso = mapmri.MapmriModel(gtab, radial_order=radial_order,
+                                                     laplacian_regularization=True,
+                                                     laplacian_weighting=laplacian_weighting)
+                mapfit_aniso = map_model_aniso.fit(data)
+
```

Comment:
Thanks for this work @aarya22. I just have a couple of questions:

- When `laplacian=True` and `posivity=True`, the first `if` is overridden by the others... there is something wrong here
- What happens when `laplacian=False` ?
- What happens when `laplacian=False` and `posivity=False` ? We never know how it will be used...

Reply:
If laplacian=True and positivity=True, the first if should override the next two if statements right? This is supposed to happen, in order to create the correct mapmri model type given the inputs.

Perhaps it is not quite clear, but either the laplacian or positivity or both must be true in order for the workflow to function. If they are both false then the workflow will not work. I can add that in the description.

Reply:
Ok, Thanks ! Maybe you should warn the users by generating a warning or print a message when both value are `False`

Reply:
Good idea! I just added that as an error message.

Reply:
Why not add a line with "elif not laplacian and not positivity" and then just set laplacian_regularization to False in the model call? There's no reason to put an exception here since the model will just do least squares in this case.

Reply:
@rutgerfick Do you mean to make a model with the following parameters in the case of not laplacian and not positivity?
 
`mapmri.MapmriModel(gtab, radial_order=radial_order, laplacian_regularization=False)`

Reply:
if positivity_constraint's default is False then that should do it :-)

Reply:
Sounds good, will make the change. @rutgerfick Thanks for the suggestion.